python/cornserve/services/task_registry/manager.py

```diff
@@ -0,0 +1,357 @@
+    async def create_task_instance_from_task(
+        self,
+        task: UnitTask,
+        task_uuid: str
+    ) -> str:
+        """Create a named task instance from a configured task object.
+
+        Returns the instance name.
+        """
+        await self._load_config()
+        assert self._custom_api is not None
+
+        task_type = task.__class__.__name__.lower()
+        instance_name = f"{task_type}-{task_uuid}"
+
+        body = {
+            "apiVersion": f"{CRD_GROUP}/{CRD_VERSION}",
+            "kind": "UnitTaskInstance",
+            "metadata": {"name": instance_name, "namespace": K8S_NAMESPACE},
+            "spec": {
+                "definitionRef": task.__class__.__name__,
+                "config": task.model_dump(mode="json"),
+                "executionDescriptorName": task.execution_descriptor_name,
+            },
+        }
+
+        try:
+            await self._custom_api.create_namespaced_custom_object(
+                group=CRD_GROUP,
+                version=CRD_VERSION,
+                namespace=K8S_NAMESPACE,
+                plural="unittaskinstances",
+                body=body,
+            )
+            logger.info("Created task instance: %s", instance_name)
+            return instance_name
+        except client.ApiException as e:
+            if e.status == 409:
+                raise ValueError(f"Task instance {instance_name} already exists") from e
+            raise
+
+    async def create_execution_descriptor(
+        self,
+        name: str,
+        task_class_name: str,
+        descriptor_class_name: str,
+        module_name: str,
+        source_code: str,
+        is_default: bool = True,
+    ) -> dict[str, Any]:
+        """Create an execution descriptor from source code."""
+        await self._load_config()
+        assert self._custom_api is not None
+
+        encoded_source = base64.b64encode(source_code.encode("utf-8")).decode("utf-8")
+
+        body = {
+            "apiVersion": f"{CRD_GROUP}/{CRD_VERSION}",
+            "kind": "ExecutionDescriptor",
+            "metadata": {"name": name, "namespace": K8S_NAMESPACE},
+            "spec": {
+                "taskClassName": task_class_name,
+                "descriptorClassName": descriptor_class_name,
+                "moduleName": module_name,
+                "sourceCode": encoded_source,
+                "isDefault": is_default,
+            },
+        }
+
+        try:
+            return await self._custom_api.create_namespaced_custom_object(
+                group=CRD_GROUP,
+                version=CRD_VERSION,
+                namespace=K8S_NAMESPACE,
+                plural="executiondescriptors",
+                body=body,
+            )
+        except client.ApiException as e:
+            if e.status == 409:
+                raise ValueError(f"Execution descriptor {name} already exists") from e
+            raise
+
+    async def get_task_instance(
+        self, instance_name: str
+    ) -> UnitTask:
```

Comment:
I thought in the event when this method was called by a Cornserve service (e.g., resource manager) too quickly and getting the CRD object returned a 404, it would be useful to do a little bit of retries, maybe two to three retries with exponential backoff. Would this situation happen? Does retry make sense?